newIDE/app/src/InstancesEditor/InstancePropertiesEditor/index.js

```diff
@@ -25,6 +25,7 @@ type Props = {|
   layout: gdLayout,
   instances: Array<gdInitialInstance>,
   onEditObjectByName: string => void,
+  onInstancesModified?: Instances => void,
```

Comment:
Actually, this is a `Array<gdInitialInstance>` that is passed to onInstancesModified - this is more precise than a `Instances`.

`Instances` is a generic type for PropertiesEditor - as PropertiesEditor can be used for anything else that would have "properties".
In theory, we would have typed PropertiesEditor as a generic component in Flow that would take a type `Instances` as parameter (a `PropertiesEditor<Instances>` if you follow me). 
And Flow would have deduced that this `Instances` type is in this case a `Array<gdInitialInstance>`.

In practice it was too hard, so we just typed this as a "Object" type (i.e: a bit like "any"):
![image](https://user-images.githubusercontent.com/1280130/135297923-85d6abc7-bc72-4bab-9698-f0bfc7c47d6c.png)

Let's avoid leaking this to the parent, and use Array<gdInitialInstance>, much like the `instances` prop that is passed down.

Reply:
Ok, thanks for the precision!